ts/src/huobi.ts

```diff
@@ -2809,120 +2898,126 @@ export default class huobi extends Exchange {
         const result = {};
-        this.options['networkChainIdsByNames'] = {};
-        this.options['networkNamesByChainIds'] = {};
-        for (let i = 0; i < data.length; i++) {
-            const entry = data[i];
-            const currencyId = this.safeString (entry, 'currency');
+        const currencyIds = Object.keys (dataByCurrencyId);
+        for (let i = 0; i < currencyIds.length; i++) {
+            const currencyId = currencyIds[i];
             const code = this.safeCurrencyCode (currencyId);
-            this.options['networkChainIdsByNames'][code] = {};
-            const chains = this.safeValue (entry, 'chains', []);
+            const chains = dataByCurrencyId[currencyId];
             const networks = {};
-            const instStatus = this.safeString (entry, 'instStatus');
-            const currencyActive = instStatus === 'normal';
+            let currencyActive = false;
+            let depositEnabled = false;
+            let withdrawEnabled = false;
             let minPrecision = undefined;
-            let minWithdraw = undefined;
-            let maxWithdraw = undefined;
-            let deposit = false;
-            let withdraw = false;
+            let isFiat = false;
             for (let j = 0; j < chains.length; j++) {
-                const chainEntry = chains[j];
-                const uniqueChainId = this.safeString (chainEntry, 'chain'); // i.e. usdterc20, trc20usdt ...
-                const title = this.safeString (chainEntry, 'displayName');
-                this.options['networkChainIdsByNames'][code][title] = uniqueChainId;
-                this.options['networkNamesByChainIds'][uniqueChainId] = title;
-                const networkCode = this.networkIdToCode (title, code);
-                minWithdraw = this.safeNumber (chainEntry, 'minWithdrawAmt');
-                maxWithdraw = this.safeNumber (chainEntry, 'maxWithdrawAmt');
-                const withdrawStatus = this.safeString (chainEntry, 'withdrawStatus');
-                const depositStatus = this.safeString (chainEntry, 'depositStatus');
-                const withdrawEnabled = (withdrawStatus === 'allowed');
-                const depositEnabled = (depositStatus === 'allowed');
-                withdraw = (withdrawEnabled) ? withdrawEnabled : withdraw;
-                deposit = (depositEnabled) ? depositEnabled : deposit;
-                const active = withdrawEnabled && depositEnabled;
-                const precision = this.parsePrecision (this.safeString (chainEntry, 'withdrawPrecision'));
-                if (precision !== undefined) {
-                    minPrecision = (minPrecision === undefined) ? precision : Precise.stringMin (precision, minPrecision);
+                const chain = chains[j];
+                const networkId = this.safeString (chain, 'chain');
+                const networkTitle = this.safeString (chain, 'ac', ''); // empty names are for some exceptional FIAT currencies
```

Comment:
@ttodua also not sure we should be using `ac` here, for instance this chain seems to be representing bep20 and not eth 
![image](https://github.com/ccxt/ccxt/assets/43336371/daacfbf7-a73e-4824-b3e2-50a53eef0828)

Reply:
![image](https://github.com/ccxt/ccxt/assets/43336371/04475123-1b30-47f2-a68f-1d6878d4c3ca)
 

`dn` seems a bit more reliable, but man I don't like this v1 endpoint such a weird response with weird fields

Reply:
ok `dn` is not consistent either,  I'm seriously considering keeping the v2 and losing some chains instead of dealing with this
![image](https://github.com/ccxt/ccxt/assets/43336371/ec72134f-11bd-4d4e-9ae2-345d9db71147)

Reply:
I think I'd better write tests for networks at first, because the discrepancies you found manually, i missed them

Reply:
okkk, i am checking the things again, if it's true and v1 is also buggy, then we should discard this PR too